CONTRIBUTING.md

```diff
@@ -324,14 +328,51 @@ Secretless supports using [Go plugins](https://golang.org/pkg/plugin/) to extend
 its functionality. To learn about writing new Secretless plugins and for more
 information on the types of plugins we currently support, visit the [plugin API directory](pkg/secretless/plugin).
 
-### Submodules
+## Submodules
 
 Secretless makes use of some plugins using [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). 
 Development on submodules is similar to just working with a second repository, in that you can `cd` into it and 
 check out branches or make seperate commits. However, you also have the ability to commit and push recursively 
 from the parent repository. For help with this, it is recommended to review the "Publishing Submodule Changes" 
 section of the Git Submodules documentation.
 
+`git submodule update --recursive` can be used to update the registered
+ submodules to match what the superproject expects by cloning missing submodules and
+ updating the working tree of the submodules. The "updating" can be done in several
+ ways depending on command line options and the value of submodule.<name>.update
+ configuration variable. The command line option takes precedence over the
+ configuration variable. If neither is given, a checkout is performed. The recursive
+  flag will handle any submodules nested within a submodule
+  
+### Updating Submodules
+When making a change to a submodule, it will not be committed automatically when the
+super repository is committed. As such, there are a few steps in place to make sure
+this happens.
+
+1. Enter the submodule directory and create a branch to store your changes, publishing
+ this branch now or once you have completed your work.
+
+2. Commit any neccessary changes within the submodule repository to your new branch. If
```

Comment:
More info on commands and the effect of pushing might be useful here (does it create a fork on the remote of the submodule repo?)